framework/src/executor/mod.rs

```diff
@@ -59,6 +58,31 @@ impl<S: Storage, DB: TrieDB, Mapping: ServiceMapping> Clone for ServiceExecutor<
 impl<S: 'static + Storage, DB: 'static + TrieDB, Mapping: 'static + ServiceMapping>
     ServiceExecutor<S, DB, Mapping>
 {
+    pub async fn create_schema(
+        trie_db: Arc<DB>,
+        storage: Arc<S>,
+        mapping: Arc<Mapping>,
+    ) -> ProtocolResult<()> {
+        let querier = Rc::new(DefaultChainQuerier::new(Arc::clone(&storage)));
+        let mut schema = vec![];
+        for name in mapping.list_service_name().iter() {
+            let trie = MPTTrie::new(Arc::clone(&trie_db));
+            let sdk = DefalutServiceSDK::new(
+                Rc::new(RefCell::new(GeneralServiceState::new(trie))),
+                Rc::clone(&querier),
+                NoopDispatcher {},
+            );
+            let service = mapping.get_service(&name, sdk)?;
+            let ret = service.schema_();
+            schema.push(ServiceSchema {
+                service: name.clone(),
+                method:  ret.0,
+                event:   ret.1,
+            });
+        }
+        storage.insert_schema(ChainSchema { schema }).await
```

Comment:
Seems like it is not necessary to insert the schema into storage. Is is possible to remain it in memory, like a global lazy static variable, and make it accessible for API module? Then we don't need to implement the `insert_schema` and `get_schema` method.

Reply:
That is really a option, and i think it twice. But there's no obvious benefits to change to that, maintaining it in memory, because we don't use schema frequently while running node.

Reply:
I think keeping it in memory is a better option, with the benefit that no matter how long the program runs, only the first call is expensive.

Reply:
The usage of schema in storage is a little weird. We insert it into storage every time we start the process, and get it when the API invoke get_schema. For me it's more like a memory thing than a storage thing. The drawback is that you have to implement `insert_schema` and `get_schema` method in storage trait.

Consider that you have already implemented it in this way, it also makes sense. The performance differs little. So it is not a big deal and pretty fine to remain it in this way.

Reply:
When you consider [this design](https://github.com/nervosnetwork/muta/pull/264#issuecomment-621670185), you will feel that it is a natural thing to keep in memory.

Reply:
Ok, i will keep it in memory